### PR TITLE
use gitignore.io for gitignore

### PR DESCRIPTION
**Pull request by pat-s**
-> https://www.toptal.com/developers/gitignore/

- Templates: `node` & `go`
- Moved custom rules to the bottom
- Ignored `node_modules/`

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

**Review by 6543 — CHANGES_REQUESTED** [review]
dublicated entrys now

**Comment by anbraten** [comment]
Was there anything missing in the current ignore?

**Comment by pat-s** [comment]
> Was there anything missing in the current ignore?

For me, I had a lot of `node_modules/` stuff in there from some frontend testing at some point. 

Overall, the use of gitignore.io simplifies rule creation - and also allows for additional custom rules. Opinionated but widely used.

File: .gitignore [hunk 1]
```diff
@@ -1,26 +1,183 @@
...
-__debug_bin*
 
 # Test binary, built with `go test -c`
 *.test
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
 
+# Dependency directories (remove the comment below to include it)
+# vendor/
```
Comment (6543) [inline .gitignore:23]:
```suggestion
# Dependency directories
vendor/
```

**Review by 6543 — APPROVED** [review]

**Comment by anbraten** [comment]
Those templates add quite some unrelated stuff (we are not using half of the stuff like npm, yarn, jest, next etc) while still missing other web related parts. Wont block however.

**Comment by pat-s** [comment]
> Those templates add quite some unrelated stuff (we are not using half of the stuff like npm, yarn, jest, next etc) 

That's how such templates work. Still way easier to use them then to initialize custom gitignore's for every project IMO. 

> while still missing other web related parts.

Well, same as above. Templates won't cover *everything* but theres the "custom" section to extend them.

Won't merge then if not everybody likes it.

**Pull request closed without merging**